### PR TITLE
[develop2] Let dependency define package_id modes

### PR DESCRIPTION
**Pull request by memsharded**

### CODE REVIEW

**Review by lasote — COMMENTED** [review]
Ok to the code & implementation but I would wait until the feedback of the Tribe. 
It could be quite messy to debug in environments like conan-center.

**Comment by memsharded** [comment]
Agree, lets move it to beta.2 then, and wait for the feedback.

**Review by lasote — APPROVED** [review]

File: conans/model/requires.py [hunk 1]
```diff
@@ -321,6 +324,10 @@ def deduce_package_id_mode(self, pkg_type, dep_pkg_type, non_embed_mode, embed_m
             self.package_id_mode = "unrelated_mode"
             return
 
+        # If the dependency defines the mode, that has priority over default
+        embed_mode = getattr(dep_conanfile, "embed_mode", embed_mode)
```
Comment (memsharded) [inline conans/model/requires.py]: Maybe the naming should be more explicit? Like ``package_id_embed_mode``?

Comment (memsharded) [reply conans/model/requires.py]: Done

**Review by lasote — APPROVED** [review]

**Review by czoido — APPROVED** [review]

**Pull request merged**